### PR TITLE
fix: When configured using a pipeline, handle null filterable

### PR DESCRIPTION
**Pull request by rahulsom**
When using pipelines, people tend to skip `filterable` as a parameter. That causes `filterable` to be set to `null`. Stapler would emit an empty string.

The prior behavior used javascript truthiness and treated it as `false`.

Since moving almost all javascript into separate files, the new behavior causes an empty string to be printed during the method call.

This change guarantees we emit a primitive `boolean` instead of a boxed `Boolean`.

Refs: [JENKINS-71724](https://issues.jenkins.io/browse/JENKINS-71724)

### Testing done

Added new tests.
Existing tests pass.

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by kinow — COMMENTED** [review]
Thanks for working on this @rahulsom ! I saw some discussion on the issues, but I am not able to follow that at the moment.

Build is failing, but the code/tests look good to me. Once the build is green I can merge/cut a release.

Thanks!

**Comment by rahulsom** [comment]
It's failing to checkout - I've started a discussion - https://groups.google.com/g/jenkinsci-dev/c/JlSMkucHulo

**Review by kinow — APPROVED** [review]
:tada:

**Comment by kinow** [comment]
Shall we merge and release this, @rahulsom ?

**Comment by rahulsom** [comment]
Yep. We're good to release this. Thanks!

**Comment by kinow** [comment]
Release started, then stopped. All tests worked, Maven build was going fine, then Jenkins maven repo rejected my credentials.

No changes on my side since last release a few weeks ago (as I'm working more on Python nowadays). Checked Maven security settings and they look good.

Logged out, then logged in again in JIRA, all good. But https://repo.jenkins-ci.org/ui/login/ rejects my credentials. Tried to recover password, and it says my user either doesn't exist or is locked.

One of the reasons I stopped contributing to Jenkins projects & plugins was the amount of things needed to stay atop of the moving parts, and the time spent troubleshooting issues like this (<rant-over/>).

I have a full day with things I will have to deliver at $work, then a medical appointment at evening. Ditto for tomorrow. Will try the `mvn release:perform` again in the evening or weekend, but if that doesn't work in the next days I might have to leave it like that until I'm back from Brazil by the end of September.

**Comment by oleg-z** [comment]
>Logged out, then logged in again in JIRA, all good. But https://repo.jenkins-ci.org/ui/login/ rejects my credentials. Tried to recover password, and it says my user either doesn't exist or is locked.
Can someone from jenkins core team help with the issue? @oleg-nenashev @jglick?

**Comment by jglick** [comment]
use https://github.com/jenkins-infra/helpdesk/issues

**Comment by jglick** [comment]
(or switch to https://www.jenkins.io/doc/developer/publishing/releasing-cd/ in which case you never need to bother with personal Artifactory credentials again)

**Comment by kinow** [comment]
That's correct @eplodn. After my last message I tried to perform the last step of the release process one last time, and the account was probably still locked or something (last time I used the credentials was for the previous release :shrug: ).

I'm busy with last minute deliveries at $work before going to visit family. I'll be back by October, and then once I find spare time I will try the jenkins-infra/helpdesk issues or switching as Jesse suggested. If the issue or switch do not take too long, a new release should be out somewhere in October.

Sorry.

**Comment by basil** [comment]
Released in [2.7.2](https://github.com/jenkinsci/active-choices-plugin/releases/tag/uno-choice-2.7.2).

**Comment by jglick** [comment]
> I have to remember to use a separate branch

Sorry, what do you mean by a separate branch? Maybe there is something that needs to be clarified in the docs.

**Comment by kinow** [comment]
> > I have to remember to use a separate branch
> 
> Sorry, what do you mean by a separate branch? Maybe there is something that needs to be clarified in the docs.

Hi @jglick , maybe I misunderstood this part

>Push the work to prepare your plugin for CD to a dedicated branch, which you file a pull request from to your default branch. When enabling CD, the hosting team reviews this PR containing all necessary changes in your plugin, to prevent improper releases and mistakes. Don’t push the changes directly to the default branch or merge the PR before the hosting team has reviewed it. 

I thought it was suggesting to use something like `release` or `develop` branches, and that pull requests would have to be made to this branch, and eventually merged into `master` or `main` to be automagically released (like in gitflow).

**Comment by jglick** [comment]
No no, not at all. The “dedicated branch” refers to _setting up_ the plugin for CD, one time. This is just saying that when following the instructions in the guide, rather than pushing all those changes straight to `master`, it is better to create a regular PR that the hosting team can review in case you missed a step or whatever. Then once `cd.enabled` is merged to trunk in RPU and the Artifactory secrets appear in your repo, you are all set to merge the CD setup PR. After that, you just merge PRs to your plugin’s default branch (e.g., `master`) normally, taking care to apply a label like `bug` or `chore` both for release notes categorization and to instruct the CD system whether a release is needed or not.

**Comment by kinow** [comment]
Aaahh! That doesn't sound too bad! I might try that in one of the next releases (and if that works will move other plugins to CD too). Thanks for clarifying @jglick !

**Pull request merged**
